src/server/api/user.ts

```diff
@@ -310,6 +310,11 @@ router.patch('/url', validateState, async (req, res) => {
         }
       })
     }
+
+    if (url.isFile) {
```

Comment:
this operation needs to be wrapped in a transaction because of two-phase commit issues.